hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -268,13 +268,11 @@ protected boolean initializeIfNeeded(HoodieTableMetaClient dataMetaClient,
       // If there is no commit on the dataset yet, use the SOLO_COMMIT_TIMESTAMP as the instant time for initial commit
       // Otherwise, we use the timestamp of the latest completed action.
       String initializationTime = dataMetaClient.getActiveTimeline().filterCompletedInstants().lastInstant().map(HoodieInstant::getTimestamp).orElse(SOLO_COMMIT_TIMESTAMP);
-
-      // Initialize partitions for the first time using data from the files on the file system
-      if (!initializeFromFilesystem(initializationTime, partitionsToInit, inflightInstantTimestamp)) {
+      initializeFromFilesystem(initializationTime, partitionsToInit, inflightInstantTimestamp);
+      if (!this.dataMetaClient.getTableConfig().isMetadataPartitionAvailable(MetadataPartitionType.FILES)) {
```

Comment:
Should this check all metadata partitions that should be initialized besides FILES?

Reply:
for other partitions, either we succeed/return true or throw. there are no conditions under which the method initializeFromFilesystem will return gracefully w/o initializing partitions.

Reply:
OK, then should we remove this if check?  Such checks on specific metadata partition always confuse me.